File: app/src/main/java/closer/vlllage/com/closer/handler/media/PlaceholderHandler.java
package closer.vlllage.com.closer.handler.media;

import android.content.res.Resources;
import android.graphics.Bitmap;
import android.graphics.BitmapFactory;
import android.graphics.drawable.BitmapDrawable;
import android.util.Base64;

import closer.vlllage.com.closer.pool.PoolMember;

public class PlaceholderHandler extends PoolMember {
    public int getHeightFromAspectRatio(float width, float aspectRatio) {
        return (int) (width / aspectRatio);
    }

    public BitmapDrawable drawableFromBase64(Resources resources, String image) {
        byte bytes[] = Base64.decode(image, Base64.DEFAULT);

        Bitmap bitmap = BitmapFactory.decodeByteArray(bytes, 0, bytes.length);
        return new BitmapDrawable(resources, bitmap);
    }

}

File: app/src/main/java/closer/vlllage/com/closer/handler/media/PlaceholderHandler.kt
package closer.vlllage.com.closer.handler.media

import android.content.res.Resources
import android.graphics.BitmapFactory
import android.graphics.drawable.BitmapDrawable
import android.util.Base64
import com.queatz.on.On

class PlaceholderHandler constructor(private val on: On) {
    fun getHeightFromAspectRatio(width: Float, aspectRatio: Float): Int {
        return (width / aspectRatio).toInt()
    }

    fun drawableFromBase64(resources: Resources, image: String): BitmapDrawable {
        val bytes = Base64.decode(image, Base64.DEFAULT)

        val bitmap = BitmapFactory.decodeByteArray(bytes, 0, bytes.size)
        return BitmapDrawable(resources, bitmap)
    }

}
